app/components/agenda/agendaitem/agendaitem-decision.js

```diff
@@ -47,6 +49,18 @@ export default class AgendaitemDecisionComponent extends Component {
     this.previousReport = yield this.report?.previousPiece;
   }
 
+  @action
+  async onSaveDecisionResultCode() {
+    const decisionResultCode = await this.args.decisionActivity.decisionResultCode;
+    if ([CONSTANTS.DECISION_RESULT_CODE_URIS.UITGESTELD, CONSTANTS.DECISION_RESULT_CODE_URIS.INGETROKKEN].includes(decisionResultCode.uri)) {
+      const pieces = await this.args.agendaitem.get('pieces');
```

Comment:
I assume `get()` is required since `this.args.agendaitem` can be a promise? Can we try to avoid to pass promise objects as arguments?

Reply:
Will change. In this case it doesn't really matter because we're just awaiting the relation. But does "avoid passing promise objects as arguments" also apply when we're e.g. using the relations in the template (there, it being a PromiseProxy is actually useful because we can then rely on it for reactivity purposes, at least that's how I understand Ember, but I could be wrong)?